src/main/java/build/buildfarm/worker/CASFileCache.java

```diff
@@ -2436,6 +2459,7 @@ private CancellableOutputStream putOrReferenceGuarded(
                     expiredFuture,
                     (expiredKey) -> {
                       try {
+                        new File(expiredKey.toString()).setWritable(true, true);
```

Comment:
Double check the behavior here, file delete (unlink) only requires perms on the dir, and java may not perform any consideration for a readonly file deletion.